### PR TITLE
assignment-11

### PR DESCRIPTION
**Pull request by lkrawczykowski**

### CODE REVIEW

File: 11/client/src/javascripts/app.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    wd.createRoot('app')
+        .routes([
+            { url: '/', templateUrl: 'routes/home.html' },
+            { url: '/sign-in', templateUrl: 'routes/sign-in.html' }
+        ])
+        .component("RouterLink", {
+            template: "<a href=\"\"><\/a>",
+            beforeMount: function (app, element, componentData) {
+                let replacement = document.createElement("a");
+                replacement.href = element.getAttribute("href");
```
Comment (Reshetnyak) [inline 11/client/src/javascripts/app.js]: Please, see my comments for task 10 pull request. The same can be improved here.

File: 11/client/src/javascripts/wd.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        },
+        routes: function (routes) {
+            this.appRoutes = routes;
+            return this;
+        },
+        navigate: function (url, affectsHistory = true) {
+            let app = this;
+            let templateUrl = this.appRoutes.find(x => x.url === url).templateUrl;
+            if (templateUrl === undefined)
+                console.error(url + " is not defined");
```
Comment (Reshetnyak) [inline 11/client/src/javascripts/wd.js]: Please, avoid such constructions. You wouldn't be able to use debugger or set a breakpoint in the body of `if`

File: 11/client/src/javascripts/wd.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    if (affectsHistory)
+                        history.pushState({ url: url }, 'route: ' + url, url);
+                }).catch(function (error) {
+                    console.error(error);
+                });
+            }
+            return this;
+        },
+        component: function (name, descriptor) {
+            this.components.push({ name: name, descriptor: descriptor });
```
Comment (Reshetnyak) [inline 11/client/src/javascripts/wd.js]: You are using ES2015 and can leverage [object properties shorthands](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer)
```javacript
this.components.push({ name, descriptor });
```

**Review by Reshetnyak — CHANGES_REQUESTED** [review]

File: 11/client/src/javascripts/wd.js (hunk 2 above)
Comment (lkrawczykowski) [reply 11/client/src/javascripts/wd.js]: I am not sure what you mean, lack of brackets?

Comment (Reshetnyak) [reply 11/client/src/javascripts/wd.js]: Exactly. I would use examples next time to not confuse you.

**Pull request merged**